### ISSUE TITLE
Beta 0.30 bug: React 19 version mismatch - TypeError: Cannot read properties of undefined (reading 'ReactCurrentDispatcher')

### ISSUE THREAD

**Issue by kodobrody** [namespace-ee/react-calendar-timeline#937]
**Describe the bug**
```
⨯ node_modules\react-calendar-timeline\dist\react-calendar-timeline.es.js (195:1) @ eval
 ⨯ TypeError: Cannot read properties of undefined (reading 'ReactCurrentDispatcher')
    at __webpack_require__ (C:\Users\adamm\react-calendar-test\.next\server\webpack-runtime.js:33:42)
    at eval (./src/app/test.tsx:7:81)
    at (ssr)/./src/app/test.tsx (C:\Users\adamm\react-calendar-test\.next\server\app\page.js:159:1)
    at Object.__webpack_require__ [as require] (C:\Users\adamm\react-calendar-test\.next\server\webpack-runtime.js:33:42)
digest: "2676660959"
  193 |       }
  194 |     }
> 195 |     var ke = V.ReactCurrentDispatcher, je;
      | ^
  196 |     function ze(g, D, H) {
  197 |       {
  198 |         if (je === void 0)
```

**To Reproduce**

1. Initialize next.js project (it defaults to react 19 and next.js 15, when tried in vite it works fine as it uses react 18)
2. Install 0.30@beta-1
3. setup basic example (I'm using date-fns for dates with `.valueOf()` to get numbers)

OR

1. Go to page
2. See an error

**Expected behavior**
It should render the component correctly

**Screenshots**
![image](https://github.com/user-attachments/assets/3326e88d-308f-42a3-a2a0-101b55aa167e)

**Library Version**
0.30 beta 1

**Comment by KlotzJesse** [comment]
@Remco4EF 

Can be fixed by correctly not bundling any react in the vite.config.mjs external option, like that:
`external: ["react", /^react\/.*/, "react-dom", /react-dom\/.*/],`

**Comment by KlotzJesse** [comment]
@kodobrody react 19 is being forced with nextjs 15 :)

you can verify why it is a problem by finding __SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner in the dist of the package in npm

The reason why it breaks is because __SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED were changed in react 19, which leads to a problem and react version mismatch, because __SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED is undefined, and nextjs15 is using react 19 rc

We actually also can just mark react-jsx-runtime also as external :)

minimum / recommended for react18/19 therefore:
external: ['react', 'react/jsx-runtime', 'react-dom', 'react-dom/client']
 
 helped at cal.com to also reduce bundle size from 20kb to 2kb :)

**Comment by Ilaiwi** [comment]
released in 0.30.0-beta.2
